In [1]:
import pandas as pd
df = pd.read_csv("../data/raw/NSCH 2024e_Topical_CAHMI_DRC (1).csv")
labels = pd.read_excel(
    "../data/raw/NSCH_2024e_Topical_Variable Labels_CAHMI_DRC.xlsx"
)
df.head()


,HEIGHT,FIPSST,STRATUM,HHID,FORMTYPE,TOTKIDS_R,TENURE,HHLANGUAGE,SC_AGE_YEARS,SC_SEX,...,nomAnxietyDep12to17_24,nomSystCareCSHCN_24,nomFlrish6mto5_24,nomFlrish6to17_24,nomFlrish6to17CSHCN_24,nomACE2more_24,smSmoking_24,smPhysAct12to17_24,smAdeqIns_24,smForgoneHC_24
0,9999.0,48,1,24000016,3,3,2,1,12,2,...,2,2,90,99,99,2,2,2,1,2
1,9990.0,48,1,24000017,1,1,2,1,5,2,...,95,95,1,90,90,2,2,90,1,2
2,9990.0,47,1,24000044,1,2,1,1,5,1,...,95,95,1,90,90,2,2,90,1,2
3,9990.0,13,2,24000054,1,2,2,1,2,2,...,95,95,1,90,90,2,2,90,1,2
4,9990.0,39,1,24000057,1,2,1,1,2,2,...,95,95,1,90,90,2,2,90,1,2


In [2]:
import pandas as pd

df = pd.read_csv("../data/raw/NSCH_2024_Survey_Data.csv")

labels = pd.read_excel(
    "../data/raw/NSCH_2024_Survey_Labels_ID.xlsx"
)


In [3]:
df.shape


(51375, 902)

In [4]:
df["HHID"].nunique()

51375

In [5]:
len(df)

51375

In [6]:
labels.head()

,Name,Label
0,HEIGHT,Child's Height (CM)
1,FIPSST,State FIPS Code
2,STRATUM,Sampling Stratum
3,HHID,Unique Household ID
4,FORMTYPE,Form Type


In [7]:
labels[labels["Label"].str.contains("mental", case=False, na=False)]


,Name,Label
38,K4Q28X04,Needed Health Care Not Received - Mental Healt...
100,K2Q36A,Developmental Delay
101,K2Q36B,Developmental Delay Currently
217,ACE8,Child Experienced - Lived with Mentally Ill
237,K2Q36C,Developmental Delay Severity Description
264,A1_MENTHEALTH,Adult 1 - Mental or Emotional Health
266,A2_MENTHEALTH,Adult 2 - Mental or Emotional Health
352,K4Q22_R,Mental Health Professional Treatment
353,TREATNEED,Mental Health Professional Treatment - Problem
526,DevDelay_24,Children who currently have developmental dela...


In [8]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 51375 entries, 0 to 51374
Columns: 902 entries, HEIGHT to smForgoneHC_24
dtypes: float64(3), int64(898), str(1)
memory usage: 353.5 MB


In [9]:
df.columns

Index(['HEIGHT', 'FIPSST', 'STRATUM', 'HHID', 'FORMTYPE', 'TOTKIDS_R',
       'TENURE', 'HHLANGUAGE', 'SC_AGE_YEARS', 'SC_SEX',
       ...
       'nomAnxietyDep12to17_24', 'nomSystCareCSHCN_24', 'nomFlrish6mto5_24',
       'nomFlrish6to17_24', 'nomFlrish6to17CSHCN_24', 'nomACE2more_24',
       'smSmoking_24', 'smPhysAct12to17_24', 'smAdeqIns_24', 'smForgoneHC_24'],
      dtype='str', length=902)

In [10]:
labels.info()

<class 'pandas.DataFrame'>
RangeIndex: 902 entries, 0 to 901
Data columns (total 2 columns):
 #   Column  Non-Null Count  Dtype
---  ------  --------------  -----
 0   Name    902 non-null    str  
 1   Label   902 non-null    str  
dtypes: str(2)
memory usage: 14.2 KB


In [11]:
labels[labels["Label"].str.contains("insurance", case=False, na=False)]

,Name,Label
162,KEEPINSADULT,How to Obtain/Keep Health Insurance as Child B...
163,CURRCOV,Health Insurance Coverage - Currently Covered ...
164,K12Q03,Health Insurance - Current/Former Employer or ...
165,K12Q04,Health Insurance - Insurance Company
166,K12Q12,Health Insurance - Government Assistance Plan
167,TRICARE,Health Insurance - TRICARE
168,K11Q03R,Health Insurance - Indian Health Service
169,HCCOVOTH,Health Insurance - Other
173,AVOIDCHG,Past 12 Months - Avoided Changing Jobs to Main...
277,K3Q20,Health Insurance - Benefits Cover Services


In [12]:
labels[labels["Label"].str.contains("poverty", case=False, na=False)]

,Name,Label
450,FPL_I1,"Family Poverty Ratio, First Implicate"
451,FPL_I2,"Family Poverty Ratio, Second Implicate"
452,FPL_I3,"Family Poverty Ratio, Third Implicate"
453,FPL_I4,"Family Poverty Ratio, Fourth Implicate"
454,FPL_I5,"Family Poverty Ratio, Fifth Implicate"
455,FPL_I6,"Family Poverty Ratio, Sixth Implicate"
468,povlev4_24,Income level based on family poverty level sta...
469,povSCHIP_24,Income level based on family poverty level sta...
780,WrkngPoorR_24,Indicator 6.5a: Children living in 'working po...


In [13]:
labels[labels["Label"].str.contains("anxiety", case=False, na=False)]

,Name,Label
84,K2Q33A,Anxiety
85,K2Q33B,Anxiety Currently
232,K2Q33C,Anxiety Severity Description
523,anxiety_24,"Children who currently have anxiety problems, ..."
547,AnxietSev_24,Parent-rated severity of child's current anxie...
892,nomAnxietyDep12to17_24,"NOM: Percent of adolescents, ages 12 through 1..."


In [15]:
labels.to_csv("../docs/NSCH_2024_Survey_Labels_ID.csv", index=False)

In [19]:
import pandas as pd

df = pd.read_csv("../data/raw/NSCH_2024_Survey_Data.csv")

print("Dataset Shape:", df.shape)
df.head()

Dataset Shape: (51375, 902)


,HEIGHT,FIPSST,STRATUM,HHID,FORMTYPE,TOTKIDS_R,TENURE,HHLANGUAGE,SC_AGE_YEARS,SC_SEX,...,nomAnxietyDep12to17_24,nomSystCareCSHCN_24,nomFlrish6mto5_24,nomFlrish6to17_24,nomFlrish6to17CSHCN_24,nomACE2more_24,smSmoking_24,smPhysAct12to17_24,smAdeqIns_24,smForgoneHC_24
0,9999.0,48,1,24000016,3,3,2,1,12,2,...,2,2,90,99,99,2,2,2,1,2
1,9990.0,48,1,24000017,1,1,2,1,5,2,...,95,95,1,90,90,2,2,90,1,2
2,9990.0,47,1,24000044,1,2,1,1,5,1,...,95,95,1,90,90,2,2,90,1,2
3,9990.0,13,2,24000054,1,2,2,1,2,2,...,95,95,1,90,90,2,2,90,1,2
4,9990.0,39,1,24000057,1,2,1,1,2,2,...,95,95,1,90,90,2,2,90,1,2


# NSCH 2024 Data Cleaning & Exploration

This notebook prepares the National Survey of Children's Health (NSCH) 2024 dataset for analysis. The objective is to assess data quality, identify missing values and duplicates, verify data types, and produce an analysis-ready dataset for studying unmet pediatric mental health needs.


In [21]:
# Dataset dimensions
print(f"Rows: {df.shape[0]:,}")
print(f"Columns: {df.shape[1]}")
df.info()
df.describe()
df.describe(include="all")
missing = df.isnull().sum()

missing = missing[missing > 0].sort_values(ascending=False)

missing.head(25)
missing_percent = (
    df.isnull()
      .mean()
      .mul(100)
      .sort_values(ascending=False)
)

missing_percent.head(25)
duplicates = df.duplicated().sum()

print(f"Duplicate Rows: {duplicates}")

Rows: 51,375
Columns: 902
<class 'pandas.DataFrame'>
RangeIndex: 51375 entries, 0 to 51374
Columns: 902 entries, HEIGHT to smForgoneHC_24
dtypes: float64(3), int64(898), str(1)
memory usage: 353.5 MB
Duplicate Rows: 0


In [22]:
df.dtypes.value_counts()

int64      898
float64      3
str          1
Name: count, dtype: int64

# Survey Code Exploration

In [23]:
# Check height values
df["HEIGHT"].value_counts(dropna=False).head(15)
# Check age values
df["SC_AGE_YEARS"].value_counts(dropna=False)
# Check sex values
df["SC_SEX"].value_counts(dropna=False)

SC_SEX
1    26374
2    25001
Name: count, dtype: int64

In [24]:
df.info()
duplicates = df.duplicated().sum()
print(duplicates)
df["HEIGHT"].value_counts().head(20)

<class 'pandas.DataFrame'>
RangeIndex: 51375 entries, 0 to 51374
Columns: 902 entries, HEIGHT to smForgoneHC_24
dtypes: float64(3), int64(898), str(1)
memory usage: 353.5 MB
0


HEIGHT
9990.00000    13841
162.56000      2336
157.48000      2010
152.39999      2002
121.92000      1949
167.64000      1936
160.02000      1841
165.10001      1806
170.17999      1633
172.72000      1506
154.94000      1424
175.25999      1310
9999.00000     1299
149.86000      1258
147.32001      1245
177.80000      1210
137.16000      1141
142.24001      1059
127.00000      1041
144.78000       935
Name: count, dtype: int64

# Height inspection


In [26]:
import pandas as pd

# Load the labels file
labels = pd.read_excel("../data/raw/NSCH_2024_Survey_Labels_ID.xlsx")

# Search every column for "HEIGHT"
labels[labels.astype(str)
       .apply(lambda x: x.str.contains("HEIGHT", case=False, na=False))
       .any(axis=1)]

,Name,Label
0,HEIGHT,Child's Height (CM)


In [27]:
df["HEIGHT"].describe()
df["HEIGHT"].sort_values().tail(20)

34044    9999.0
47393    9999.0
47525    9999.0
34168    9999.0
12912    9999.0
47398    9999.0
34336    9999.0
34319    9999.0
12890    9999.0
25836    9999.0
47520    9999.0
47434    9999.0
8593     9999.0
8629     9999.0
47498    9999.0
8657     9999.0
25903    9999.0
1909     9999.0
1944     9999.0
0        9999.0
Name: HEIGHT, dtype: float64

In [28]:
sorted(df["HEIGHT"].dropna().unique())[-20:]


[np.float64(147.32001),
 np.float64(149.86),
 np.float64(152.39999),
 np.float64(154.94),
 np.float64(157.48),
 np.float64(160.02),
 np.float64(162.56),
 np.float64(165.10001),
 np.float64(167.64),
 np.float64(170.17999),
 np.float64(172.72),
 np.float64(175.25999),
 np.float64(177.8),
 np.float64(180.34),
 np.float64(182.88),
 np.float64(185.42),
 np.float64(187.96001),
 np.float64(193.75999),
 np.float64(9990.0),
 np.float64(9999.0)]

In [29]:
df["HEIGHT"].dropna().unique()

array([9999.     , 9990.     ,  137.16   ,  175.25999,  139.7    ,
        162.56   ,  152.39999,  182.88   ,  160.02   ,  170.17999,
        144.78   ,  167.64   ,  103.93   ,  180.34   ,  187.96001,
        177.8    ,  147.32001,  149.86   ,  165.10001,  114.3    ,
        142.24001,  132.08   ,  121.92   ,  111.76   ,  172.72   ,
        127.     ,  154.94   ,  109.22   ,  134.62   ,  157.48   ,
        129.53999,  124.46   ,  193.75999,  119.38   ,  185.42   ,
        116.84   ])

In [30]:
sorted(df["HEIGHT"].dropna().unique())

[np.float64(103.93),
 np.float64(109.22),
 np.float64(111.76),
 np.float64(114.3),
 np.float64(116.84),
 np.float64(119.38),
 np.float64(121.92),
 np.float64(124.46),
 np.float64(127.0),
 np.float64(129.53999),
 np.float64(132.08),
 np.float64(134.62),
 np.float64(137.16),
 np.float64(139.7),
 np.float64(142.24001),
 np.float64(144.78),
 np.float64(147.32001),
 np.float64(149.86),
 np.float64(152.39999),
 np.float64(154.94),
 np.float64(157.48),
 np.float64(160.02),
 np.float64(162.56),
 np.float64(165.10001),
 np.float64(167.64),
 np.float64(170.17999),
 np.float64(172.72),
 np.float64(175.25999),
 np.float64(177.8),
 np.float64(180.34),
 np.float64(182.88),
 np.float64(185.42),
 np.float64(187.96001),
 np.float64(193.75999),
 np.float64(9990.0),
 np.float64(9999.0)]

# Define Analysis Set

In [31]:
analysis_vars = [
    # Outcomes
    "MentHCare_24",
    "MentHDiff_24",

    # Demographics
    "SC_AGE_YEARS",
    "SC_SEX",
    "SC_RACE_R",
    "HISPANIC",

    # Socioeconomic
    "povlev4_24",
    "FPL_IF",
    "TENURE",
    "ACE1",
    "FOODSIT",
    "MISSMORTGAGE",
    "EVERHOMELESS",

    # Access to Care
    "PrevMed_24",
    "K4Q20R",
    "K4Q04_R",
    "HOWMUCH",

    # Mental Health
    "ADHDMed_24",
    "ADHDBehTreat_24",

    # School
    "GRADES",
    "MAKEFRIEND",

    # Neighborhood
    "K10Q30",
    "K10Q31",
    "K10Q40_R",
    "K10Q41_R",

    # Family
    "K8Q11",
    "K8Q21",
    "K8Q30"
]

In [32]:
analysis_df = df[analysis_vars].copy()

print(analysis_df.shape)
analysis_df.head()

KeyError: "['HISPANIC'] not in index"

In [33]:
missing_vars = [col for col in analysis_vars if col not in df.columns]

print("Variables not found:")
for col in missing_vars:
    print(col)

Variables not found:
HISPANIC


# Race / Ethnicity

In [38]:
[col for col in df.columns if "RACE" in col.upper()]

['SC_RACE_R',
 'SC_RACE_R_IF',
 'SC_RACER',
 'race4_24',
 'raceAsia_24',
 'race7_24']

In [36]:
[col for col in df.columns if "HISP" in col.upper()]

['SC_HISPANIC_R', 'SC_HISPANIC_R_IF', 'hisplang_24']

In [39]:
[col for col in df.columns if "DEPR" in col.upper()]

['depress_24', 'DepresSev_24']

# Mental Health

In [40]:
[col for col in df.columns if "MENT" in col.upper()]

['APPOINTMENT',
 'A1_MENTHEALTH',
 'A2_MENTHEALTH',
 'MentHCare_24',
 'MentHDiff_24',
 'appointment_24',
 'mentor_24',
 'EmploymentSt_24',
 'npmMentHTx12to17_24',
 'npmMentor12to17_24']

In [41]:
[col for col in df.columns if "ANXI" in col.upper()]

['anxiety_24', 'AnxietSev_24', 'nomAnxietyDep12to17_24']

In [42]:
[col for col in df.columns if "DEPR" in col.upper()]

['depress_24', 'DepresSev_24']

# Poverty

In [43]:
[col for col in df.columns if "POV" in col.upper()]

['povlev4_24', 'povSCHIP_24']

In [44]:
[col for col in df.columns if "FPL" in col.upper()]

['FPL_IF', 'FPL_I1', 'FPL_I2', 'FPL_I3', 'FPL_I4', 'FPL_I5', 'FPL_I6']

# Insurance 

In [45]:
[col for col in df.columns if "INS" in col.upper()]

['KEEPINSADULT',
 'GAINSKILLS',
 'CURRINS',
 'INSGAP',
 'INSTYPE',
 'CurrIns_24',
 'InsGap_24',
 'instype_24',
 'instype3_24',
 'InsAdeq_24',
 'insurance_24',
 'DentInstruct_24',
 'GainSkill_24',
 'InsNeeds_24',
 'SmkInside_24',
 'HousingInstab_24',
 'WCRIdomains_24',
 'npmHousingInstab_24',
 'smAdeqIns_24']

# ACEs

In [46]:
[col for col in df.columns if "ACE" in col.upper()]

['ACE3',
 'ACE4',
 'ACE5',
 'ACE6',
 'ACE7',
 'ACE8',
 'ACE9',
 'ACE10',
 'ACE11',
 'DRAWAFACE',
 'ACE1',
 'SC_RACE_R',
 'SC_RACE_R_IF',
 'SC_RACER',
 'race4_24',
 'raceAsia_24',
 'race7_24',
 'DrawFace_24',
 'ACEct_24',
 'ACE2more_24',
 'ACEincome_24',
 'ACEincome2_24',
 'ACEdivorce_24',
 'ACEdeath_24',
 'ACEjail_24',
 'ACEdomviol_24',
 'ACEneighviol_24',
 'ACEmhealth_24',
 'ACEdrug_24',
 'ACEdiscrim_24',
 'ACEHealthDiscrim_24',
 'ACE6ctHH_24',
 'ACE2more6HH_24',
 'ACEctComm_24',
 'ACE1moreComm_24',
 'nomACE2more_24']

## Variable Discovery

The NSCH dataset contains 902 variables, so only variables relevant to the research question will be selected for analysis.

This section searches the dataset for variables related to:

- Mental health care and access
- Anxiety and depression
- Poverty and household resources
- Insurance coverage
- Race and ethnicity
- Adverse Childhood Experiences (ACEs)
- School, family, and neighborhood factors

Exact dataset column names are verified before creating the analysis dataset.

In [47]:
def search_variables(keyword):
    """
    Return dataset column names containing a keyword.

    Parameters
    ----------
    keyword : str
        Word or partial word to search for in column names.
    """
    matches = [
        column
        for column in df.columns
        if keyword.upper() in column.upper()
    ]

    print(f"Found {len(matches)} variables containing '{keyword}':\n")

    for column in matches:
        print(column)

    return matches

### Mental Health

In [48]:
mental_health_vars = search_variables("MENT")

Found 10 variables containing 'MENT':

APPOINTMENT
A1_MENTHEALTH
A2_MENTHEALTH
MentHCare_24
MentHDiff_24
appointment_24
mentor_24
EmploymentSt_24
npmMentHTx12to17_24
npmMentor12to17_24


### Anxiety

In [49]:
anxiety_vars = search_variables("ANXI")

Found 3 variables containing 'ANXI':

anxiety_24
AnxietSev_24
nomAnxietyDep12to17_24


### Depression

In [50]:
depression_vars = search_variables("DEPR")

Found 2 variables containing 'DEPR':

depress_24
DepresSev_24


### ADHD 

In [52]:
[col for col in df.columns if "ADHD" in col.upper()]

['ADHD_24',
 'ADHDSev_24',
 'ADHDind_24',
 'ADHDSevInd_24',
 'ADHDMed_24',
 'ADHDBehTreat_24']

In [53]:
adhd_vars = search_variables("ADHD")

Found 6 variables containing 'ADHD':

ADHD_24
ADHDSev_24
ADHDind_24
ADHDSevInd_24
ADHDMed_24
ADHDBehTreat_24


### Poverty

In [54]:
poverty_vars = search_variables("POV")
fpl_vars = search_variables("FPL")

Found 2 variables containing 'POV':

povlev4_24
povSCHIP_24
Found 7 variables containing 'FPL':

FPL_IF
FPL_I1
FPL_I2
FPL_I3
FPL_I4
FPL_I5
FPL_I6


### Insurance

In [55]:
insurance_vars = search_variables("INS")

Found 19 variables containing 'INS':

KEEPINSADULT
GAINSKILLS
CURRINS
INSGAP
INSTYPE
CurrIns_24
InsGap_24
instype_24
instype3_24
InsAdeq_24
insurance_24
DentInstruct_24
GainSkill_24
InsNeeds_24
SmkInside_24
HousingInstab_24
WCRIdomains_24
npmHousingInstab_24
smAdeqIns_24


### Race and Ethnicity

In [56]:
race_vars = search_variables("RACE")
hispanic_vars = search_variables("HISP")

Found 6 variables containing 'RACE':

SC_RACE_R
SC_RACE_R_IF
SC_RACER
race4_24
raceAsia_24
race7_24
Found 3 variables containing 'HISP':

SC_HISPANIC_R
SC_HISPANIC_R_IF
hisplang_24


In [ ]:
hispanic_vars = search_variables("HISP")

### Adverse Childhood Experiences

Adverse Childhood Experiences, or ACEs, are potentially stressful or traumatic experiences that may affect a child's health and well-being.

The NSCH dataset contains individual ACE items and derived summary measures, including counts and indicators for two or more ACEs.

In [57]:
ace_vars = search_variables("ACE")

Found 36 variables containing 'ACE':

ACE3
ACE4
ACE5
ACE6
ACE7
ACE8
ACE9
ACE10
ACE11
DRAWAFACE
ACE1
SC_RACE_R
SC_RACE_R_IF
SC_RACER
race4_24
raceAsia_24
race7_24
DrawFace_24
ACEct_24
ACE2more_24
ACEincome_24
ACEincome2_24
ACEdivorce_24
ACEdeath_24
ACEjail_24
ACEdomviol_24
ACEneighviol_24
ACEmhealth_24
ACEdrug_24
ACEdiscrim_24
ACEHealthDiscrim_24
ACE6ctHH_24
ACE2more6HH_24
ACEctComm_24
ACE1moreComm_24
nomACE2more_24


In [58]:
ace_vars = [
    column
    for column in df.columns
    if column.upper().startswith("ACE")
    or column.upper().startswith("NOMACE")
]

print(f"Found {len(ace_vars)} ACE-related variables:\n")

for column in ace_vars:
    print(column)

Found 28 ACE-related variables:

ACE3
ACE4
ACE5
ACE6
ACE7
ACE8
ACE9
ACE10
ACE11
ACE1
ACEct_24
ACE2more_24
ACEincome_24
ACEincome2_24
ACEdivorce_24
ACEdeath_24
ACEjail_24
ACEdomviol_24
ACEneighviol_24
ACEmhealth_24
ACEdrug_24
ACEdiscrim_24
ACEHealthDiscrim_24
ACE6ctHH_24
ACE2more6HH_24
ACEctComm_24
ACE1moreComm_24
nomACE2more_24


### School

In [59]:
school_vars = search_variables("SCHOOL")
attendance_vars = search_variables("ATTEND")
bully_vars = search_variables("BULL")

Found 3 variables containing 'SCHOOL':

STARTSCHOOL
SchoolReadiness_24
nomSchoolReadiness2_24
Found 0 variables containing 'ATTEND':

Found 8 variables containing 'BULL':

BULLIED_R
BULLY
bully_24
bullied_24
npmBully12to17_24
npmBully12to17CSHCN_24
npmBullied12to17_24
npmBullied12to17CSHCN_24


### Family

In [60]:
family_vars = search_variables("FAMILY")
parent_vars = search_variables("PARENT")

Found 1 variables containing 'FAMILY':

FAMILY_R
Found 3 variables containing 'PARENT':

ParentPhH_24
ParentMH_24
ParentHSt_24


### Neighborhood & Community

In [61]:
neighborhood_vars = search_variables("NEIGH")
community_vars = search_variables("COMM")
safety_vars = search_variables("SAFE")

Found 1 variables containing 'NEIGH':

ACEneighviol_24
Found 5 variables containing 'COMM':

DrCommR_24
OtherComm_24
SatOthComm_24
ACEctComm_24
ACE1moreComm_24
Found 2 variables containing 'SAFE':

NbhdSafe_24
SchlSafe_24


### Sleep

In [62]:
sleep_vars = search_variables("SLEEP")

Found 3 variables containing 'SLEEP':

HOURSLEEP05
HOURSLEEP
HrsSleep_24


### Physical Activity

In [63]:
exercise_vars = search_variables("EXER")
activity_vars = search_variables("ACTIVE")
sports_vars = search_variables("SPORT")

Found 2 variables containing 'EXER':

ENGAGE_EXERCISE
OverExercise_24
Found 3 variables containing 'ACTIVE':

A1_ACTIVE
A2_ACTIVE
ActiveWork_24
Found 3 variables containing 'SPORT':

TRANSPORTCC
transport_24
sports_24


### Gneral Health

In [64]:
health_vars = search_variables("HEALTH")

Found 8 variables containing 'HEALTH':

HEALTHKNOW
A1_PHYSHEALTH
A1_MENTHEALTH
A2_PHYSHEALTH
A2_MENTHEALTH
mhealth_24
ACEmhealth_24
ACEHealthDiscrim_24


### Special Health Care Needs

In [65]:
special_health_vars = search_variables("SHCN")
special_needs_vars = search_variables("SPECIAL")

Found 19 variables containing 'SHCN':

TOTCSHCN
TOTNONSHCN
SC_CSHCN
CSHCN_24
CSHCNtype_24
CSHCNComplex_24
CSHCNExpanded_24
CSHCNComplexExp_24
npmMedHomeCSHCN_24
npmPerDrNsCSHCN_24
npmUsualSckCSHCN_24
npmFamCentCSHCN_24
npmNoRefPrb2CSHCN_24
npmCareCoorRCSHCN_24
npmTransitionCSHCN_24
npmBully12to17CSHCN_24
npmBullied12to17CSHCN_24
nomSystCareCSHCN_24
nomFlrish6to17CSHCN_24
Found 0 variables containing 'SPECIAL':



### Household

In [66]:
household_vars = search_variables("HOUSE")
housing_vars = search_variables("HOUS")

Found 1 variables containing 'HOUSE':

HOUSE_GEN
Found 4 variables containing 'HOUS':

HOUSE_GEN
HousingInstab_24
housing_24
npmHousingInstab_24


# Create the Analysis-Ready Dataset

The full NSCH dataset contains 902 variables and children ages 0–17. This project focuses on school-aged children ages 6–17 and variables relevant to unmet pediatric mental health needs.

The cleaned dataset will retain:

- Survey design and identification variables
- Child demographics
- Mental health conditions and severity
- Mental health care access
- Insurance and healthcare access
- Poverty and material hardship
- Adverse Childhood Experiences
- Parent and family characteristics
- School and peer experiences
- Neighborhood conditions
- Sleep, physical activity, and screen time
- Special healthcare needs

Official NSCH derived measures are used whenever possible. Special survey codes representing legitimate skips, children outside a question’s age range, suppressed responses, or missing responses will be converted to missing values.

In [ ]:
selected_vars = [

    # ==========================================
    # Survey Design Variables
    # ==========================================

    "HHID",
    "STRATUM",
    "fwc",

    # ==========================================
    # Geography & Built Environment
    # ==========================================

    # Geographic Identifiers
    "FIPSST",
    "MPC_YN",
    "hrsareg",

    # Neighborhood Environment
    "NbhdSupp_24",
    "NbhdSafe_24",
    "NbhdAmenities_24",

    # Community Environment
    "OtherComm_24",
    "DrCommR_24",

    # Built Environment
    "SideWlks_24",
    "park_24",
    "RecCentr_24",
    "library_24",

    # ==========================================
    # Demographics
    # ==========================================

    "SC_AGE_YEARS",
    "SC_SEX",
    "SC_RACE_R",
    "SC_HISPANIC_R",

    # ==========================================
    # Mental Health & Neurodevelopmental Conditions
    # ==========================================

    # Anxiety
    "anxiety_24",
    "AnxietSev_24",

    # Depression
    "depress_24",
    "DepresSev_24",

    # ADHD
    "ADHD_24",
    "ADHDSev_24",
    "ADHDind_24",
    "ADHDSevInd_24",

    # Autism Spectrum Disorder
    "autism_24",
    "AutismSev_24",
    "AutismInd_24",

    # Behavioral / Conduct Problems
    "behavior_24",
    "BehavSev_24",

    # Tourette Syndrome
    "tourette_24",
    "TouretSev_24",

    # Developmental Conditions
    "DevDelay_24",
    "speech_24",
    "learning_24",
    "IntDisab_24",

    # ==========================================
    # Mental Health Treatment & Healthcare Access
    # ==========================================

    # ADHD Treatment
    "ADHDMed_24",
    "ADHDBehTreat_24",

    # Autism Treatment
    "ASDMed_24",
    "ASDBehTreat_24",

    # General Mental Health Treatment
    "MedEmotion_24",
    "therapy_24",
    "rxmeds_24",

    # Mental Health Care Access
    "MentHCare_24",
    "MentHDiff_24",

    # ==========================================
    # Mental Health Indicators & Composite Measures
    # ==========================================

    "mhealth_24",
    "mhneeds_24",
    "nomAnxietyDep12to17_24",
    "MEDB10ScrQ5_24",

    # ==========================================
    # Socioeconomic Status
    # ==========================================

    "povlev4_24",
    "povSCHIP_24",
    "FPL_IF",

    # ==========================================
    # Insurance & Healthcare Access
    # ==========================================

    "insurance_24",
    "CurrIns_24",
    "InsGap_24",
    "InsAdeq_24",

    # ==========================================
    # Adverse Childhood Experiences (ACEs)
    # ==========================================

    "ACEct_24",
    "ACE2more_24",

    # ==========================================
    # Family & Household Environment
    # ==========================================

    "FAMILY_R",
    "ParentPHH_24",
    "ParentMH_24",
    "ParentHSt_24",

    # ==========================================
    # School Readiness & Development
    # ==========================================

    "SchoolReadiness_24",

    "EL_domain_24",
    "SE_domain_24",
    "SR_domain_24",
    "M_domain_24",
    "H_domain_24",

    "flrish6to17_24",
    "flrsh6to17ct",

    # ==========================================
    # School Experience & Academic Performance
    # ==========================================

    "SchlEngage_24",
    "cares_24",
    "GRADES",
    "ReptGrade_24",
    "SchlMiss_24",

    # ==========================================
    # School Social Environment
    # ==========================================

    "bullied_24",
    "bully_24",
    "SchlSafe_24",

    # ==========================================
    # Extracurricular & Community Participation
    # ==========================================

    "AftSchAct_24",
    "sports_24",
    "clubs_24",
    "lessons_24",
    "volunteer_24",
    "EventPart_24",

    # ==========================================
    # Housing Stability
    # ==========================================

    "HousingInstab_24",

    # ==========================================
    # Physical Health & Lifestyle
    # ==========================================

    "HrsSleep_24",
    "ActiveWork_24",

    # ==========================================
    # Special Health Care Needs
    # ==========================================

    "CSHCN_24",
    "CSHCNComplex_24",

]